backend/battle/forms.py

```diff
@@ -20,13 +27,43 @@ def __init__(self, *args, **kwargs):
         super(BattleForm, self).__init__(*args, **kwargs)
         self.fields['creator'].widget = forms.HiddenInput()
 
+    def clean_opponent(self):
+        opponent_email = self.cleaned_data["opponent"]
+        try:
+            opponent = User.objects.get(email=opponent_email)
+        except User.DoesNotExist:
+            self.is_guest = True
```

Comment:
I'm not sure if this is completely safe to do.
Instead try the following: set `opponent.is_guest = True` and read new custom attr this below on `clean`.

Reply:
No need for a migration, this is just a "temporary" attr.

Reply:
But the user can be a guest or not, how can I verify this on clean?

Reply:
set `opponent.is_guest = False` above after `opponent = User.objects.get(email=opponent_email)`.

Reply:
Okay :)